src/redux/books/bookSlice.js

```diff
@@ -1,37 +1,73 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
 
 const initialState = {
-  books: [
-    {
-      item_id: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item3',
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    },
-  ],
+  books: [],
 };
 
+export const fetchBooks = createAsyncThunk('books/fetchBooks', async () => {
+  const response = await fetch('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/Q2vJ7mN9IENuKzD4CjC7/books');
+  const data = response.json();
+  return data;
+});
+
+export const addBook = createAsyncThunk('books/addBook', async ({ book }) => {
+  const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/Q2vJ7mN9IENuKzD4CjC7/books';
+  const dat = {
+    item_id: book.item_id,
+    title: book.title,
+    author: book.author,
+    category: book.category,
+  };
+  const response = await fetch(url, {
+    method: 'post',
+    headers: { 'Content-type': 'application/json' },
+    body: JSON.stringify(dat),
+  });
+  response.text();
+  return { dat };
+});
+
+export const removeBook = createAsyncThunk('books/removeBook', async ({ itemId }) => {
+  const response = await fetch(`https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/Q2vJ7mN9IENuKzD4CjC7/books/${itemId}`,
```

Comment:
- [ ] Again here, It is a requirement  that Axios is used to make the HTTP requests, kindly do so

Reply:
the instruction said we should:
Use either [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make API calls

Reply:
![Screenshot (27)](https://user-images.githubusercontent.com/83312939/222929330-0eddc283-35fe-4eef-96d7-5fcf4a1530d5.png)